python_modules/dagster/dagster/_core/blueprints/base_asset_blueprint.py

```diff
@@ -0,0 +1,83 @@
+from abc import abstractmethod
+from typing import AbstractSet, Any, Mapping, Optional, Sequence
+
+from dagster import MaterializeResult, asset, multi_asset
+from dagster._core.blueprints.blueprint import Blueprint, BlueprintDefinitions
+from dagster._core.execution.context.compute import AssetExecutionContext
+from dagster._core.models import DagsterModel
+
+
+class BaseAssetBlueprint(Blueprint):
+    """A blueprint for an asset definition whose materialization function is a databricks task."""
+
+    asset_key: str
+    deps: Sequence[str] = []
+    description: Optional[str] = None
+    metadata: Mapping[str, Any] = {}
+    group_name: Optional[str] = None
```

Comment:
We could use [field descriptions](https://docs.pydantic.dev/latest/api/fields/#pydantic.fields.Field) like we do in configurable resources, which we could lift into the JSON schema (for editor) and into our schema format (for UI) - can always do this down the line. Slight bias for these over (or in addition to) `Attributes` in docstring which is harder to machine-parse.

```python
from pydantic import BaseModel, Field


class Model(BaseModel):
    foo: str = Field(..., description='foo description', alias='FOO')
```

Reply:
Good point. Switching.